use gtest expect in ProcessGroupGlooTest instead of other checks

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32242 [flaky-tests] skip testExceptions in ProcessGroupGloo if built with TSAN
* **#32138 use gtest asserts in ProcessGroupGlooTest instead of other checks**
* #32134 [flaky-tests] fix testSend and testRecv in ProcessGroupGlooTest

I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

rohan-varma committed:
use gtest asserts in ProcessGroupGlooTest instead of other checks

I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 7614d555:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=7614d555a110192c8a07e941e98dad08ff2c1e18).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4226133) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=84513518) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=84513518))
<details>
<summary>
<code>Jan 13 22:48:11 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff</code>
</summary>

```
Jan 13 22:48:11 + XDIR=/var/lib/jenkins/workspace/xla/scripts/.. 
Jan 13 22:48:11 + PTDIR=/var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 13 22:48:11 + python /var/lib/jenkins/workspace/xla/scripts/cond_patch.py /var/lib/jenkins/workspace/xla/scripts/../torch_patches /var/lib/jenkins/workspace/xla/scripts/../.. 
Jan 13 22:48:11 1 out of 1 hunk FAILED 
Jan 13 22:48:11 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 13 22:48:11 Traceback (most recent call last): 
Jan 13 22:48:11   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 13 22:48:11     patch_repo(args) 
Jan 13 22:48:11   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 13 22:48:11     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 13 22:48:11 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 13 22:48:11 + cleanup 
Jan 13 22:48:11 + retcode=1 
Jan 13 22:48:11 + set +x 
Jan 13 22:48:11 =================== sccache compilation log =================== 
Jan 13 22:48:11 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 13 22:48:11 Compile requests                 1 
Jan 13 22:48:11 Compile requests executed        0 
Jan 13 22:48:11 Cache hits                       0 
Jan 13 22:48:11 Cache misses                     0 
Jan 13 22:48:11 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

rohan-varma committed:
Update on "use gtest asserts in ProcessGroupGlooTest instead of other checks"


I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

[ghstack-poisoned]

rohan-varma committed:
Update on "use gtest asserts in ProcessGroupGlooTest instead of other checks"


I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

[ghstack-poisoned]

rohan-varma committed:
Update on "use gtest asserts in ProcessGroupGlooTest instead of other checks"


I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

[ghstack-poisoned]

rohan-varma committed:
Update on "use gtest asserts in ProcessGroupGlooTest instead of other checks"


I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

[ghstack-poisoned]

jjlilley commented on torch/lib/c10d/test/ProcessGroupGlooTest.cpp:
```diff
@@ -185,9 +185,7 @@ void testAllreduce(const std::string& path, const at::DeviceType b) {
-      if (data[j] != expected) {
-        throw std::runtime_error("BOOM!");
-      }
+      ASSERT_EQ(data[j], expected);
```
Curious: does it make sense to use EXPECT_EQ() instead?

(ASSERT_ is fine too, though I've typically seen EXPECT_ preferred when the error doesn't affect the program control flow, so we can continue and see the other errors).

jjlilley replied:
But, to be clear, the change LGTM as-is.

rohan-varma replied:
Makes sense, I'll change it to expect.

jjlilley commented:
(*) But, to be clear, the change LGTM as-is.

rohan-varma committed:
Update on "use gtest asserts in ProcessGroupGlooTest instead of other checks"


I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

[ghstack-poisoned]

rohan-varma committed:
Update on "use gtest asserts in ProcessGroupGlooTest instead of other checks"


I personally prefer `throw std::runtime_error("BOOM")`, but we should
probably have asserts here now that it is gtest. Also ensures that the correct
exceptions are thrown by the `testSignal` tests.

Differential Revision: [D19382905](https://our.internmc.facebook.com/intern/diff/D19382905/)

[ghstack-poisoned]
